typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -141,12 +145,30 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
         await warpRoute[method]();
         if (tokenType === TokenType.collateral) {
           const wrappedToken = await warpRoute.wrappedToken();
-          const xerc20 = IXERC20__factory.connect(wrappedToken, this.provider);
           try {
+            const xerc20 = IXERC20__factory.connect(
+              wrappedToken,
+              this.provider,
+            );
             await xerc20['mintingCurrentLimitOf(address)'](warpRouteAddress);
             return TokenType.XERC20;
             // eslint-disable-next-line no-empty
           } catch {}
+
+          try {
+            const fiatToken = IFiatToken__factory.connect(
+              wrappedToken,
+              this.provider,
+            );
+
+            // Simulate minting tokens from the warp route contract
+            await fiatToken.callStatic.mint(NON_ZERO_SENDER_ADDRESS, 1, {
+              from: warpRouteAddress,
+            });
+
+            return TokenType.collateralFiat;
+            // eslint-disable-next-line no-empty
+          } catch {}
```

Comment:
Do we always ignore exceptions in calls to contracts? Or should we at least check the exception type to come from `ethers` and re-throw unexpected exceptions (I'm not sure if this is possible)?